plugins/GSdx/GSTextureCache.h

```diff
@@ -59,8 +59,9 @@ class GSTextureCache
 		void Flush(uint32 count, int layer);
 
 	public:
+		void* m_palette_obj; // Pointer to the relevant Palette object (if any)
```

Comment:
The problem is that the definition of `Palette` happens later in the code, with respect to that member definition, so it cannot be referenced yet, but I can pre-define an empty Palette class to avoid compilation error, though.

Reply:
I think it's also possible to rearrange the order so you don't have to forward declare Palette.